File: Array/linchuanxin/Algorithm/day24_0624_环形链表/day_24_leetcode_141.c
//
// Created by LCX on 2020/6/24.
//

#include <stdio.h>
#include <stdbool.h>


 typedef struct ListNode {
      int val;
      struct ListNode *next;
 }Node;

bool hasCycle(Node *head) {
    Node *fast = head;
    Node *slow = head;

    while (fast != NULL)
    {
        fast = fast->next;
        if(fast != NULL)
        {
            fast = fast->next;
        }
        if(fast == slow)
        {
            return true;
        }
        slow = slow->next;
    }

    return false;
}

File: Array/linchuanxin/Algorithm/day24_0624_环形链表/day24_leetcode_141.kt
package day24


class ListNode(var `val`: Int) {
    var next: ListNode? = null
}



fun hasCycle(head: ListNode?): Boolean {
    var fast = head
    var slow = head

    while (fast != null){
        fast = fast.next
        if(fast != null){
            fast = fast.next
        }
        if(fast == slow){
            return true
        }
//        if(slow != null){
//            slow = slow.next
//        }
        slow = slow?.next
    }
    return false
}


fun main() {

}